portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +40,18 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
     // adds comments to datastore 
     String comment = request.getParameter("comments");
+    String nickname = getUserNickname(userService.getCurrentUser().getUserId());
```

Comment:
This is fine for now, but if you ever want to come back to refactor this, I would suggest making a separate NicknameServlet (to handle user nickname setting), and renaming this one to the CommentServlet.

since this servlet updates to comments here, this handler should be a doPost (posts are used for writes). Fetching comments without writes should use a doGet instead.